### PR TITLE
[Armadactl] Authenticate via token from external cmd

### PR DESCRIPTION
**Pull request by d80tb7**
### What changes were proposed in this pull request?
Adds a new type of auth `ExecAuth` to ArmadaCtl that causes an external program to be invoked to generate a BeraerToken.

### Why are the changes needed?
Allows Armada to integrate with external auth systems for which it doesn't have native support

### Does this PR introduce any user-facing change?
Yes.  Users can now modify their Armada Configuration to include an ExecAuth Section.  For Example,
```
execAuth:
  cmd: some-command

  args:
    - "arg1"

  env:
    - name: "FOO"
      value: "bar"

  interactive: true
```
The stdout from the command will be used as a bearer token when making calls to armada server.
### How was this patch tested?
Unit tests

### CODE REVIEW

File: pkg/client/auth/exec/authenticator.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
+package exec
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	grpc_credentials "google.golang.org/grpc/credentials"
```
Comment (carlocamurri) [inline pkg/client/auth/exec/authenticator.go:7]: Looks like this is out of order, not sure why it's not being picked up by `goimports`

File: pkg/exec.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package main
```
Comment (carlocamurri) [inline pkg/exec.go:1]: Is this file needed?

Comment (d80tb7) [reply pkg/exec.go]: ah yes good spot

File: pkg/client/auth/exec/authenticator.go (hunk 1 above)
Comment (d80tb7) [reply pkg/client/auth/exec/authenticator.go]: done

**Review by carlocamurri — APPROVED** [review]
